reports/agent-that-syncs-pods-and-google-sheets.md

```diff
@@ -0,0 +1,178 @@
+<!--
+Fill in the WebIDs of the people below.
+Leave this in comments!
+It's possible to have multiple people per role.
+
+Challenge/scenario creator:
+  - https://pieterheyvaert.com/#me
+Solution creator:
+  - https://pieterheyvaert.com/#me
+Report writer:
+  - https://pieterheyvaert.com/#me
+-->
+
+# Agent that syncs pods and Google Sheets
+
+The corresponding challenge is [#120](https://github.com/SolidLabResearch/Challenges/issues/120),
+which contributes to scenario [#119](https://github.com/SolidLabResearch/Challenges/issues/119).
+
+## Problem
+<!--
+You can reuse the pitch of the challenge, but check if you need to make changes.
+For example, it might happen that the approved solution does more than what the original pitch requested.
+-->
+
+A lot of users use Google Sheets to view and edit data. 
+This data can be stored in Google Sheets itself or 
+it can be coming from an external data source such as a 
+[MySQL database](https://coefficient.io/how-to-connect-mysql-database-to-google-sheets) or 
+[MongoDB](https://hightouch.com/integrations/google-sheets-source-to-mongodb). 
+So far, there is no solution that exists that allows doing this with data in Solid pods. 
+Challenge [#120](https://github.com/SolidLabResearch/Challenges/issues/120) 
+focuses on the agent/backend part of a solution for this problem.
+
+## Approved solution
+<!--
+Provide information about the approved solution:
+names of tools/libraries created, repos, and so on.
+-->
+
+We created an [agent](https://github.com/SolidLabResearch/google-sheet-sync/releases/tag/v1.0.0) that 
+syncs data between a Google Sheet and a resource on a Solid pod.
+It has the following features:
+
+- Users define which Google Sheet and resource the agent syncs.
+- Users define via a SPARQL query what data the agent selects from the resource.
+- Users define via YARRRML rules how the data from the sheet is converted to RDF before it is stored in the resource.
+- The agent overwrites the sheet with all the queried data when it launches.
+- The agent periodically checks the sheet for changes and stores these changes in the resource.
+- The agent only works with public resources.
+
+You find a screencast of the agent [here](https://cloud.ilabt.imec.be/index.php/s/eFrEKF2YCkSx22j).
+
+<!--
+Provide a list of important technical decisions and assumptions.
+-->
+We made the following important technological decisions and assumptions:
```

Comment:
I'd like to know how you figured out which rows were changed. Does each row have an id assigned? Otherwise it's impossible to know when row 2 was deleted and at the same time a new row was added before row 3, bc without ID you'll assume this is an edit of row 2, even though it might be about totally different resources?

Reply:
I'd also like to know how you adapt this code to your own use case, could you add a link to https://github.com/SolidLabResearch/google-sheet-sync/blob/main/README.md#configuration ?